### PR TITLE
Fix anyof and oneof inside additional properties.

### PR DESCRIPTION
**Pull request by sempostma**
### Reasons for making this change

oneOf and anyOf properties are not working properly inside additional properties. 

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by sempostma** [comment]
@epicfaace  I have added the tests in https://github.com/rjsf-team/react-jsonschema-form/pull/1588/commits/9be4b1a4d15ba9afa980045a46c5cc6e10b217a1.

**Comment by sempostma** [comment]
@epicfaace Is this still awaiting review?

File: packages/core/src/components/fields/ObjectField.js [hunk 1]
```diff
@@ -83,7 +83,7 @@ class ObjectField extends Component {
 
   onPropertyChange = (name, addedByAdditionalProperties = false) => {
     return (value, errorSchema) => {
-      if (!value && addedByAdditionalProperties) {
+      if (typeof value === "undefined" && addedByAdditionalProperties) {
```
Comment (epicfaace) [inline packages/core/src/components/fields/ObjectField.js:86]: This change seems unrelated to the oneOf change -- could you add another test to cover it? (I believe this fixes an issue in which you can't type in "0" in a oneOf textbox, I think)

Comment (epicfaace) [inline packages/core/src/components/fields/ObjectField.js:86]: Let's do this, because the variable `value` is always going to be defined (whereas `value`'s value may be equal to undefined)

```suggestion
      if (value === undefined && addedByAdditionalProperties) {
```

File: packages/core/src/utils.js [hunk 2]
```diff
@@ -201,11 +201,21 @@ function computeDefaults(
       )
     );
   } else if ("oneOf" in schema) {
-    schema =
-      schema.oneOf[getMatchingOption(undefined, schema.oneOf, rootSchema)];
+    const matchingKey = getMatchingOption(undefined, schema.oneOf, rootSchema);
+    if (schema.oneOf.hasOwnProperty(matchingKey) === false) {
+      // oneOf is an empty array so there is no schema available.
```
Comment (epicfaace) [inline packages/core/src/utils.js:206]: Can you add tests that cover this case as well (and update the PR description)?

File: packages/core/src/utils.js [hunk 3]
```diff
@@ -201,11 +201,21 @@ function computeDefaults(
       )
     );
   } else if ("oneOf" in schema) {
-    schema =
-      schema.oneOf[getMatchingOption(undefined, schema.oneOf, rootSchema)];
+    const matchingKey = getMatchingOption(undefined, schema.oneOf, rootSchema);
+    if (schema.oneOf.hasOwnProperty(matchingKey) === false) {
```
Comment (epicfaace) [inline packages/core/src/utils.js:205]: Would it be clearer to instead check if `schema.oneOf.length === 0`, so we don't have to unnecessarily call `getMatchingOption`?

**Comment by heath-freenome** [comment]
@memalloc Given the changes in v5 beta, the code here needs refactoring. Is this something you are willing to do?

**Comment by heath-freenome** [comment]
Closing this PR as it is reimplemented in #3435
